### ISSUE TITLE
Don't bring food to an empty mine

### ISSUE THREAD

**Issue by HoraK-FDF** [freeserf/freeserf#146]
Hi,

if an mine is empty it still gets food supply that is than used for nothing and had been better used in an active mine. as everywhere wasting food is not good ;) 

... greetings HoraK-FDF

**Comment by jarrodmoldrich** [comment]
My 2 cents... there's an over-arching philosophy to the game that it leaves you with enough rope to hang yourself with.  Another philosophy is that eschews god-driven AI for automaton behavior.  As the serf agent can only report that they have seen nothing the X times they went down, it has no knowledge of what really is in the nearby rocks.  The only time this philosophy is breached is the starting super-geologist.  An up-side to this is that it's easy to scale, in terms of parallel processing, networking, and coding effort.  It would make AI a great deal more modular, and scaling would permit simulation and breeding of artificial intelligence via genetic and monte-carlo methods.  It would have been extra important on 7mhz 68000 machines trying to simulate 1000s of serfs.

I also think the old-school 'rogue-like' mechanics are one of the more charming aspects of this slow-moving game, which I found to be part meditation, part war strategy.

**Comment by panzerchris** [comment]
I agree with @jarrodmoldrich's thoughts on this to be honest. It's up to the player to manage their settlement effectively and there are many times (despite it being unlikely) where I have seen a mine come back to life even if for a short duration. That's why I'm still sending the miner food and not burning it down... my optimism that the miner might actually have found something again. At most, the only change would be to have a re-alert where if after another 15 the minter still finds nothing, you're informed again.

**Comment by tlongstretch** [comment]
I never understood how the mining worked at that level in the original game... so the miner is actually "travelling" to nearby tiles randomly and if that tile has the target material it is gathered, but otherwise not?  And as time goes by the chance of hitting a target material drops.

Does this also mean that two mines of the same type right near each other only serves to gather the same pool of materials faster, but does not increase the total materials gathered?  

Also... doesn't this mean that if you build a mine in an area, let it run dry, then send geologists back to the same place they should only be finding "empty flags" or maybe small deposits?  

I have always placed mines on all the large resource flags in an area, but only connecting one to the road system, thus leaving the unbuilt mines as markers where large deposits were, then connecting them once the first mine ran dry.  If I'm understanding this right, what I've been doing is unproductive and the subsequent mines in the same area should not be gathering any significant materials.  I never paid enough attention to notice.  

And finally, this means you can race competing players to mine the resources in an area before they can.   All this makes sense but I had never considered that it works this way.

**Comment by tlongstretch** [comment]
I found some information in a FAQ that seems to contradict this explanation:

"3.04) Depleting ore veins
Q. Once you excavate all of one resource from a spot, is that spot
useless for any other mining?  I could swear that at the beginning
of my game, one area showed "Very Much" in all of the resources.
Then after I exhausted the gold, the geologists planted blank signs
in all those spots.

A. Basically, yes.  Each spot only contains one type of ore and they become
depleted, although I have been able to get up to 25% output from a second
mine at the same spot (geologist told me there was more ore).  I don't
trust the assessments at the beginning of the game, since they never seem
to be completely accurate with the findings of my geologists.

3.05) Can you deplete nearby ore veins?
Q. If you set up one mine and exhaust that spot, do you also exhaust
the surrounding sectors?

A. No.  I've mined out one spot, destroyed the mine and surrounding paths
and then been able to mine in nearby locations."  

http://stanislavs.org/OldPages/stanislavs/games/serfcity.faq

**Comment by wdigger** [comment]
> tI never understood how the mining worked at that level in the original game... so the miner is actually "traveling" underground to nearby tiles randomly and if that tile has the target material it is gathered, but otherwise not? And as time goes by the chance of hitting a target material drops.
> 
> Does this also mean that two mines of the same type right near each other only serves to gather the same pool of materials faster, but does not increase the total materials gathered?
> 
> Also... doesn't this mean that if you build a mine in an area, let it run dry, then send geologists back to the same place they should only be finding "empty flags" or maybe small deposits?
> 
> I have always placed mines on all the large resource flags in an area, but only connecting one to the road system, thus leaving the unbuilt mines as markers where large deposits were, then connecting them once the first mine ran dry. If I'm understanding this right, what I've been doing is unproductive and the subsequent mines in the same area should not be gathering any significant materials. I never paid enough attention to notice.
> 
> And finally, this means you can race competing players to mine the resources in an area before they can. All this makes sense but I had never considered that it works this way.
> 
> Oh and one more thing... In my AI code I'm heavily preferring building mines on large flags over small ones... but if mining works this way it doesn't matter so much and the flags just serve as general guidelines to help find the center of a deposit.

You are right mainly, but radius of finding resource on ground is small enough. Search area - 32 points around the mine. It's about 3 points in each direction.

I'm not sure, but the restriction of construction near the border should impede competition. Between the mines can compete, but this is simply avoided on the basis of the estimated area. You can find similar checks by symbol `pos_add_spirally`, the maximum offset for this case is 32.

**Comment by tlongstretch** [comment]
Thanks, wdigger.
